packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -447,26 +437,22 @@ export const SampleAssayDetailImpl: FC<Props & InjectedAssayModel> = props => {
         }, {});
 
         // keep tab when "all" view has data, but filtered view is blank
-        const includeUnfilteredConfigs =
-            showAliquotViewSelector && activeSampleAliquotType && activeSampleAliquotType !== ALIQUOT_FILTER_MODE.all;
-        if (includeUnfilteredConfigs) {
-            const _unfilteredConfigs = getSampleAssayQueryConfigs(
-                assayModel,
-                allSampleIds,
-                queryGridSuffix,
-                UNFILTERED_GRID_ID_PREFIX,
-                false,
-                sampleSchemaQuery
-            );
+        const _unfilteredConfigs = getSampleAssayQueryConfigs(
+            assayModel,
+            allSampleIds,
+            queryGridSuffix,
+            UNFILTERED_GRID_ID_PREFIX,
+            false,
+            sampleSchemaQuery
+        );
 
-            const unfilteredConfigs = _unfilteredConfigs.reduce((_configs, config) => {
-                const modelId = config.id;
-                _configs[modelId] = config;
-                return _configs;
-            }, {});
+        const unfilteredConfigs = _unfilteredConfigs.reduce((_configs, config) => {
```

Comment:
In the case there is no aliquot filter, but only user filters, the set of filtered and unfiltered configs are identical. So the effect is to hold 2 copies of queryModel, one that user can filter on, one that's the original row. sample/assays and source/assays grids are already expensive. This change would double the amount of queries for the default case, which is when no filters are applied. 

What we really need is getting the row count before a user filter is applied. Is there a better way to achieve this?